web/pgadmin/utils/__init__.py

```diff
@@ -781,14 +781,15 @@ def clear_database_servers(load_user=current_user, from_setup=False):
     for server in servers:
         db.session.delete(server)
 
-    # Remove all groups
-    groups = ServerGroup.query.filter_by(user_id=user_id)
+    # Remove all servergroups except for the first
+    # This matches the UI behavior in
+    # web/pgadmin/browser/server_groups/__init__.py#delete
```

Comment:
When two pieces/places in the code need to match, I sometimes find it helpful to make a similar comment/reference at the other end.